src/main/java/entity_gateway/TodoEntityManager.java

```diff
@@ -0,0 +1,65 @@
+package main.java.entity_gateway;
+
+import main.java.entity.Task;
+import main.java.entity.TodoList;
+import main.java.use_case.TodoListTaskCreationModel;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class TodoEntityManager implements TodoListManager{
+    private final List<Task> taskArrayList;
+    TodoList todoList;
+    int taskCounter;
+    int todoCounter;
+
+    public TodoEntityManager(){
+        taskArrayList = new ArrayList<>();
+        taskCounter = 0;
+        todoCounter = 0;
+    }
+
+    @Override
+    public int addTask(TodoListTaskCreationModel taskData) {
+        String name = taskData.getName();
+        Duration duration = taskData.getDuration();
+        LocalDateTime deadline = taskData.getDeadline();
+        List<String> subtasks = taskData.getSubtasks();
+
+        taskCounter++;
+        Task task = new Task(name, duration, deadline, subtasks, taskCounter);
+
+        taskArrayList.add(task);
+        return taskCounter;
+    }
+
+    @Override
+    public int createTodoList() {
+        todoList = new TodoList();
+        todoCounter++;
+        return todoCounter;
+    }
+
+    @Override
+    public TaskReader getTask(int todoListId, int taskId){
+        for (Task t : taskArrayList)
+            if (t.getId() == taskId)
+                return new TaskToTaskReader(t);
+        return null;
+    }
+
+    @Override
+    public Map<Integer, List<TaskReader>> getAllTasks() {
```

Comment:
If you meant why return a HashMap instead of a Map, that's because ideally pretty much all explicit types we work with should be abstract. The main exceptions I can think off are the Main (the dirtiest of all components) and some Simple Factories.
If you mean why are we returning this mapping at all, it's just because the previous code used it and I mainly designed this PR to restructure the classes, not to delete or refactor any of the previous approaches.

Reply:
I meant why it returns a map of list and I was wondering why we would have several lists of `TaskReader`.